Library/Homebrew/env_config.rb

```diff
@@ -0,0 +1,298 @@
+        default:      HOMEBREW_DEFAULT_TEMP,
+      },
+      HOMEBREW_UPDATE_TO_TAG:             {
+        description: "Instructs Homebrew to always use the latest stable tag (even if developer commands " \
+                     "have been run).",
+        boolean:     true,
+      },
+      HOMEBREW_VERBOSE:                   {
+        description: "Homebrew always assumes `--verbose` when running commands.",
+        boolean:     true,
+      },
+      HOMEBREW_VERBOSE_USING_DOTS:        {
+        boolean:     true,
+        description: "Homebrew's verbose output will print a `.` no more than once a minute. This can be " \
+                     "useful to avoid long-running Homebrew commands being killed due to no output.",
+      },
+      HOMEBREW_BINTRAY_USER:              {
+        description: "Homebrew uses this username when accessing the Bintray API (where bottles are stored).",
+      },
+      HOMEBREW_BINTRAY_KEY:               {
+        description: "Homebrew uses this API key when accessing the Bintray API (where bottles are stored).",
+      },
+      http_proxy:                         {
+        description: "Sets the HTTP proxy to be used by `curl`(1), `git`(1) and `svn`(1) when downloading " \
+                     "through Homebrew.",
+      },
+      https_proxy:                        {
+        description: "Sets the HTTPS proxy to be used by `curl`(1), `git`(1) and `svn`(1) when downloading " \
+                     "through Homebrew.",
+      },
+      all_proxy:                          {
+        description: "Sets the SOCKS5 proxy to be used by `curl`(1), `git`(1) and `svn`(1) when downloading " \
+                     "through Homebrew.",
+      },
+      ftp_proxy:                          {
+        description: "Sets the FTP proxy to be used by `curl`(1), `git`(1) and `svn`(1) when downloading " \
+                     "through Homebrew.",
+      },
+      no_proxy:                           {
+        description: "Sets the comma-separated list of hostnames and domain names that should be excluded " \
+                     "from proxying by `curl`(1), `git`(1) and `svn`(1) when downloading through Homebrew.",
+      },
+    }.freeze
+
+    def env_method_name(env, hash)
+      method_name = env.to_s
+                       .sub(/^HOMEBREW_/, "")
+                       .downcase
+      method_name = "#{method_name}?" if hash[:boolean]
+      method_name
+    end
+
+    ENVS.each do |env, hash|
+      method_name = env_method_name(env, hash)
+      env = env.to_s
+
+      if hash[:boolean]
+        define_method(method_name) do
+          ENV[env].present?
+        end
+      elsif hash[:default].present?
+        # Needs a custom implementation.
+        next if env == "HOMEBREW_MAKE_JOBS"
+
+        define_method(method_name) do
+          ENV[env].presence || hash.fetch(:default).to_s
+        end
+      else
+        define_method(method_name) do
+          ENV[env].presence
+        end
```

Comment:
It might be possible to condense this like:
```suggestion
      else
        # Needs a custom implementation.
        next if env == "HOMEBREW_MAKE_JOBS"

        define_method(method_name) do
          ENV[env].presence || hash[:default].to_s.presence
        end
```

Or am I overcomplicating it/breaking something?

It has a slight behavioural change of allowing `default: false` or `default: []`, but I don't think that matters.

Reply:
I'd rather the more explicit version here as I think it's a valid separate "path" through the code (and will be tested separately accordingly)